pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/segment/strategy/SegmentAssignmentStrategyFactory.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.controller.helix.core.assignment.segment.strategy;
+
+import java.util.Map;
+import org.apache.pinot.common.assignment.InstancePartitions;
+import org.apache.pinot.spi.config.table.TableConfig;
+import org.apache.pinot.spi.config.table.TableType;
+import org.apache.pinot.spi.config.table.assignment.InstancePartitionsType;
+import org.apache.pinot.spi.config.table.assignment.SegmentAssignmentConfig;
+import org.apache.pinot.spi.utils.CommonConstants.Segment.AssignmentStrategy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Factory for SegmentAssignmentStrategy
+ */
+public class SegmentAssignmentStrategyFactory {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SegmentAssignmentStrategyFactory.class);
+  private SegmentAssignmentStrategyFactory() {
+  }
+
+  /**
+   * Determine Segment Assignment strategy
+   */
+  public static SegmentAssignmentStrategy getSegmentAssignmentStrategy(TableConfig tableConfig,
```

Comment:
let's improve this factory. Currently, it can only generate SegmentAssignmentStrategy for `Completed` or `OFFLINE`.

Eventually, we should create `OFFLINE` for offline tables and create both `CONSUMING/COMPLETED` for realtime tables.

We can do one of 2 approaches:

1. make the return type to be `Map<String, SegmentAssignmentStrategy>` and properly return 1 offline strategy for offline tables and 2 strategies (consuming, completed) for realtime tables. (here, you can add `TODO` comment that you will be addressing `consuming` segments in the next PR. You can just add `COMPLETED` to the map for now.

2. another approach is to keep this factory to be very simple (i.e. just pass the strategyName `e.g. AssignmentStrategy.BALANCE_NUM_SEGMENT_ASSIGNMENT_STRATEGY `) and return the strategy object. Then, we can add all the complexity for choosing the correct strategy in `setSegmentAssignmentStrategyMap()` in `OfflineSegmentAssignment/RealtimeSegmentAssignment`

Reply:
updated